horizon/data_manager/policy_store.py

```diff
@@ -0,0 +1,139 @@
+from typing import Optional
+
+from aiohttp import ClientSession
+from loguru import logger
+from opal_client.policy_store.opa_client import OpaClient
+from opal_client.policy_store.schemas import PolicyStoreAuth
+from opal_common.schemas.data import JsonableValue
+
+
+class DataManagerPolicyStoreClient(OpaClient):
+    def __init__(
+        self,
+        data_manager_client: ClientSession,
+        opa_server_url=None,
+        opa_auth_token: Optional[str] = None,
+        auth_type: PolicyStoreAuth = PolicyStoreAuth.NONE,
+        oauth_client_id: Optional[str] = None,
+        oauth_client_secret: Optional[str] = None,
+        oauth_server: Optional[str] = None,
+        data_updater_enabled: bool = True,
+        policy_updater_enabled: bool = True,
+        cache_policy_data: bool = False,
+        tls_client_cert: Optional[str] = None,
+        tls_client_key: Optional[str] = None,
+        tls_ca: Optional[str] = None,
+    ):
+        super().__init__(
+            opa_server_url,
+            opa_auth_token,
+            auth_type,
+            oauth_client_id,
+            oauth_client_secret,
+            oauth_server,
+            data_updater_enabled,
+            policy_updater_enabled,
+            cache_policy_data,
+            tls_client_cert,
+            tls_client_key,
+            tls_ca,
+        )
+        self._client = data_manager_client
+
+    async def set_policy_data(
+        self,
+        policy_data: JsonableValue,
+        path: str = "",
+        transaction_id: Optional[str] = None,
+    ):
+        parts = path.lstrip("/").split("/")
+        match parts:
+            case ["relationship_tuples", subject]:
+                for full_relation, targets in policy_data.items():
+                    relation = full_relation.lstrip("relation:")
+                    for target_type, target_objects in targets.items():
+                        for target in target_objects:
+                            # TODO missing subject_type and target_type
+                            await self._insert_fact(
+                                "relationships",
+                                {
+                                    "subject": subject,
+                                    "relation": relation,
+                                    "object": target,
+                                    "tenant": "",  # TODO unnecessary?
```

Comment:
IMO opal data should definitely send the tenant information and we should sync it as part of the fact.
but not a blocker.

Reply:
Relationship does not have a tenant, logically. The tenant association is for either of the resource instances at either end.

Reply:
i disagree but we can sort that out later when we decide on the schema.